pkg/core/managers/apis/mesh/mesh_helpers.go

```diff
@@ -33,3 +34,16 @@ func CreateDefaultMesh(resManager core_manager.ResourceManager, template mesh_pr
 
 	return nil
 }
+
+func EnsureCAs(ctx context.Context, caManagers core_ca.Managers, mesh *mesh_core.MeshResource, meshName string) error {
+	for _, backend := range mesh.Spec.GetMtls().GetBackends() {
+		caManager, exist := caManagers[backend.Type]
+		if !exist { // this should be caught by validator earlier
+			return errors.Errorf("CA manager for type %s does not exist", backend.Type)
+		}
+		if err := caManager.Ensure(ctx, meshName, *backend); err != nil {
```

Comment:
That's a bit strange to me, why do we want to ensure that all backends are fine and have CA? Why not just the active one?

Reply:
I feel like if you declare a CA as one of the backends you want to create it and `enabledBackend` is just a switcher between them. It the same as you validate all of them even that only one is active. Does it make sense?

If you still feel like we should change it I can make it happen, such change should not break anything.

Reply:
After discussing it offline, I agree that we should only ensure enabled backend. Ensure essentially checks if the CA is available (ex. provided cert is not expired or Vault PKI is in place).

Changed.